test/caching_test.rb

```diff
@@ -0,0 +1,61 @@
+require File.expand_path('../support/test_helper', __FILE__)
+
+class CachingTestTest < Minitest::Test
+  URI = "http://example.com/test-schema.json"
+
+  def test_caching
+    initial_schema = {
+      "type" => "string"
+    }
+    schema = JSON::Schema.new(initial_schema, URI)
+    JSON::Validator.add_schema(schema)
+    assert_valid(URI, "foo", :clear_cache => false)
+
+    initial_schema = {
+      "type" => "number"
+    }
+    schema = JSON::Schema.new(initial_schema, URI)
+    JSON::Validator.add_schema(schema)
+    assert_valid(URI, "foo")
```

Comment:
This would be clearer if it used refute_valid and used 123 as data (not a string)